### PR TITLE
add basic css support

### PR DESCRIPTION
**Pull request by snellcode**
very basic css support

### CODE REVIEW

**Comment by longrunningprocess** [comment]
this could also use an update to the README if you don't mind @snellcode .  I think I'll start using yours instead since it's got the basic css support that everyone wants and is requested in #10 .

**Comment by snellcode** [comment]
OK, will take a look at README.

**Comment by longrunningprocess** [comment]
@bengsfort any chance we could request you take a couple 30 min from dungeon crawler for this? ;-)  thanks!

**Comment by snellcode** [comment]
Not sure what to say. If there are files with .css ext, it will catch those and use <link tag for them. A test that confirms this happens may be the best thing, and/or a full demo folder that shows the entire usage.

Basically, I don't understand what you expect from me, but happy to help if I can.

**Comment by longrunningprocess** [comment]
hmmm, I haven't seen this plugin do the same thing for css that it does for js...

If you take a look at @snellcode 's code, you'll see that it simply injects the bundled css the same way it does for the js, right now the only way I know to accomplish this is with the replace...

```
// src/index.template.html
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset='utf-8'>
	<meta name='viewport' content='width=device-width,initial-scale=1'>

	<title>Svelte app</title>
	
	<link rel='icon' type='image/png' href='/favicon.png'>

	<link rel='stylesheet' href='bundle.[timestamp].css'>
</head>

<body>
</body>
</html>
```

```
// rollup.config.js
...
const timestamp = Date.now()

export default {
...
		htmlTemplate({
			attrs: ['defer'],
			replaceVars: {
				'[timestamp]': timestamp,
			},
			target: 'dist/index.html',
      template: 'src/index.template.html',
		}),
...
```

**Comment by longrunningprocess** [comment]
resulting in 

```
// dist/index.html
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset='utf-8'>
	<meta name='viewport' content='width=device-width,initial-scale=1'>

	<title>Svelte app</title>
	
	<link rel='icon' type='image/png' href='/favicon.png'>

	<link rel='stylesheet' href='bundle.1592325691370.css'>
</head>

<body>
<script defer src="bundle.1592325691370.js"></script>
</body>
</html>
```

**Comment by snellcode** [comment]
for reference, this is the project I am using this for:
https://github.com/snellcode/rollup-preact-starter

the tsx support is not working yet, but I commented that out for now. It's not relevant to this issue. The css build part should be working. If you have a css file provided to rollup as in this example, you should be getting the dist/index.html with the css link in the head section.

**Comment by longrunningprocess** [comment]
yep @snellcode , that worked perfectly.  No pressure @bengsfort id you're not into merging the PR it's no biggie, @snellcode 's is doing it right.  We could just use his if he wouldn't mind publishing a version to npm...

File: src/index.js [hunk 1]
```diff
@@ -45,17 +46,33 @@ export default function htmlTemplate(options = {}) {
...
+
+          // Inject the style tags before the head close tag
+          const headCloseTag = injected.lastIndexOf("</head>");
+
+          // Inject the script tags before the body close tag
+          injected = [
+            injected.slice(0, headCloseTag),
+            ...bundleKeys.filter(f => path.extname(f) === '.css').map(
+              b =>
+                `<link rel="stylesheet" type="text/css" href="${prefix || ""}${b}">\n`
```
Comment (longrunningprocess) [inline src/index.js:60]: fwiw, I don't think `type="text/css"` is needed.

**Comment by bengsfort** [comment]
@snellcode in regards to readme, he was saying to add an update to the readme about how basic css support works so that people know how to get it working without looking at code or anything.

@longrunningprocess any PR's that add functionality or add a potential for breaking bugs need a test to validate that it can't break the plugin since the plugin is used quite frequently. I can go through the code and try to write tests for them myself, but it will not be merged tonight.

**Comment by bengsfort** [comment]
I'd love to get this merged and published, I just want to make sure there are no regressions or bugs introduced, and unfortunately I don't any more development time in me today.

**Comment by longrunningprocess** [comment]
that's understandable @bengsfort, no problem at all, I can definitely wait for it.  Thanks for taking it on, really appreciate it!

**Comment by snellcode** [comment]
@bengsfort understood about readme, but here's my offer. You ask me whatever you need, and when you are happy with all of this and have time, you can update the readme as is appropriate for your approved (and maybe edited) implementation.

**Comment by longrunningprocess** [comment]
take a look at this @snellcode :  https://github.com/snellcode/rollup-plugin-generate-html-template/pull/1

**Comment by longrunningprocess** [comment]
Good catch, that's better!

**Comment by bengsfort** [comment]
Merged, will publish once tests are done

**Comment by bengsfort** [comment]
@longrunningprocess @snellcode this has now been published as `v1.7.0`. Thanks!

**Comment by longrunningprocess** [comment]
Thanks @bengsfort , I know this is going to help a bunch of people!

**Pull request merged**